src/main/java/edu/cornell/kfs/coa/batch/service/impl/CreateClosedAccountsCsvServiceImpl.java

```
@@ -0,0 +1,303 @@
+        }
+    }
+    
+    private void ensureOutbundDirectoriesExist() {
+        try {
+            FileUtils.forceMkdir(new File(getCsvClosedAccountstFileCreationDirectory()));
+        } catch (IOException e) {
+            LOG.info("ensureOutbundDirectoriesExist: Could not make file creation directory => " + getCsvClosedAccountstFileCreationDirectory() + " Throwing RuntimeException to force batch job failure.");
+            throw new RuntimeException(e);
+        }
+        try {
+            FileUtils.forceMkdir(new File(getCsvClosedAccountsExportDirectory()));
+        } catch (IOException e) {
+            LOG.info("ensureOutbundDirectoriesExist: Could not make file export directory => " + getCsvClosedAccountsExportDirectory() + " Throwing RuntimeException to force batch job failure.");
+            throw new RuntimeException(e);
+        }
+    }
+    
+    private void writeClosedAccountsToCsvFile(List<ClosedAccount> closedAccountsDataList) throws IOException {
+        String csvFileName = generateCsvOutputFileName();
+        String fullyQualifiedCreationDirectoryFileName = fullyQualifyFileNameToCreationDirectory(csvFileName);
+        LOG.info("writeClosedAccountsToCsvFile: fullyQualifiedOutputFile = " + fullyQualifiedCreationDirectoryFileName);
+        File outputFile = new File(fullyQualifiedCreationDirectoryFileName);
+        FileWriter outputFileWriter;
+        outputFileWriter = new FileWriter(outputFile);
+        try {
+            for (ClosedAccount closedAccount : closedAccountsDataList) {
+                outputFileWriter.write(closedAccount.toCsvString());
+                outputFileWriter.write(KFSConstants.NEWLINE);
+            }
+        } catch (IOException io) {
+            LOG.error("writeClosedAccountsToCsvFile: Caught IOException attempting to create CSV file of closded accounts => ", io);
+            throw io;
+        } finally {
+            outputFileWriter.flush();
+            outputFileWriter.close();
```

You may want to keep the flush() call at the end of the "try" block for simpler error handling. Also, it would be good to attempt the close() call in a way that will ignore any errors it throws, such as by using the IOUtils.closeQuietly() method.